text/trace/0000-error_flagging.md

```diff
@@ -0,0 +1,86 @@
+ 
+### Status Codes
+The following status codes are added to our current  schema.
+ 
+* `DEFAULT` No status has been set. Any errors must be detected by the analysis tool. (This replaces `OK`.)
+* `ERROR` Instrumentation has marked a span as an error. (This replaces `UNKNOWN`.)
+* `OK_OVERRIDE` The user has provided an override. The span should NOT be flagged as an error, regardless of other analysis.
+* `ERROR_OVERRIDE` The user has provided an override. The span SHOULD be flagged as an error, regardless of other analysis.
+ 
+(Note that our current status codes include a long list of error types. We may choose to keep  them, change them, or drop them in favor of a single `ERROR` code. How many error types we have is not relevant to this proposal.)
+ 
+`OK_OVERRIDE` and `ERROR_OVERRIDE` are special status codes. These are explicit overrides provided by the end user, and should never be set by shared instrumentation. They should only be set by the application developer (via application code), or by the operator (via the collector).
+ 
+Analysis tools are free to disregard status codes, in favor of their own approach to error analysis. However it is strongly suggested that analysis tools handle `OK_OVERRIDE` and `ERROR_OVERRIDE`, as these are explicitly set by the end user and contain valuable information.
+ 
+ 
+### Error Mapping Schema
+As part of the specification, OpenTelemetry provides a canonical mapping of semantic conventions to status codes. This removes any ambiguity as to what OpenTelemetry ships with out of the box.
+ 
+### Error Processor
+The collector will provide a processor and a configuration language to make adjustments to this error mapping schema. This provides the flexibility and customization needed for real world scenarios.
+ 
+### Convenience methods
+As a convenience, OpenTelemetry provides helper functions for adding semantic conventions and exceptions to a span. These helper functions will also set the correct status code. This simplifies the life of the instrumentation author, and helps ensure compliance and data quality.
+
+Note that these convenience methods simply wire together multiple API calls. They should live in a helper package, and should not be directly added to existing API interfaces. Given how many semantic conventions we have, there will be a pile of them.
+ 
+ 
+## Internal details
+Except for the renaming of two status codes, this proposal is backwards compatible with existing code, protocols, and the OpenTracing bridge.
+ 
+**OK is renamed to DEFAULT**
+ Using the term “OK” as the default implies more meaning than we intend. The span is not necessarily OK - it simply has not triggered our standard error mapping. The default status code should be renamed to `DEFAULT` to avoid confusion.
+ 
+ Note: I intentionally avoided terms like "unset" as it may imply to users that they are required to set the status code, which is not the intention.
```

Comment:
If we don't want to use the term `UNSET`, because the _set_ verb implies that the user should be doing some setting, perhaps something like `NONE` or `NULL`.

Reply:
NO_OVERRIDE would make perfect sense to me

Reply:
All gone now!